### PR TITLE
[release-1.0] cherrypick 1846, 1996

### PR DESCRIPTION
**Pull request by lzhecheng**
#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
[release-1.0] cherrypick 1846, 1996
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/network/private_link_service.go [hunk 1]
```diff
@@ -68,13 +68,13 @@ var _ = Describe("Private link service", Label(utils.TestSuiteLabelPrivateLinkSe
...
 		utils.Logf("Creating deployment " + serviceName)
-		deployment := createNginxDeploymentManifest(serviceName, labels)
+		deployment := createServerDeploymentManifest(serviceName, labels)
 		_, err = cs.AppsV1().Deployments(ns.Name).Create(context.TODO(), deployment, metav1.CreateOptions{})
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	AfterEach(func() {
-		err := cs.AppsV1().Deployments(ns.Name).Delete(context.TODO(), serviceName, metav1.DeleteOptions{})
+		err := utils.DeletePod(cs, ns.Name, utils.ExecAgnhostPod)
```
Comment (jwtty) [inline tests/e2e/network/private_link_service.go:77]: looks like here we need to delete the deployment

Comment (lzhecheng) [reply tests/e2e/network/private_link_service.go]: Updated.

**Comment by MartinForReal** [comment]
<html>
<body>


/test pull-cloud-provider-azure-e2e-ccm-vmss-basic-lb-1-21
--


<br class="Apple-interchange-newline">
</body>
</html>

**Pull request merged**